### PR TITLE
dockerized version of user-templates-api

### PR DESCRIPTION
**Pull request by gesinaphillips**
This PR represents code that successfully creates a functional docker container.

Questions:
- ~~This currently uses port 5050 which is on Zhou's list of ports currently in use. There is also a reference to port 80 in passthrough.py. Advice on choosing ports?~~
- ~~I'm also not totally sure about the function of passthrough.py overall.~~
- ~~qcluster is complaining about redis, but I wanted to make sure qcluster is necessary here before fixing. If user_workspaces_server is a good model, it looks like I'd need to add redis back to the docker-compose file as well as potentially config.json.~~
- ~~Django(?) is complaining about the ASGI lifespan protocol, so I've shut that off in start.sh, but I am not positive that's a great solution.~~

This PR does not include linting. That will be a separate PR.

### CODE REVIEW

**Comment by jpuerto-psc** [comment]
- Passthrough.conf can be removed. This was used in user-workspaces to proxy users to their interactive session
- Let's use port 5001

File: docker/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,14 @@
+FROM python:3.8-buster
+WORKDIR /code
+ENV PYTHONUNBUFFERED=1
+# TODO: added RUN to create tmp dir but not sure if needed,
+# or if RUN on line 9 should just install in current dir
+RUN /bin/bash -c 'mkdir tmp'
+COPY requirements.txt /tmp/
+RUN pip install --upgrade pip
+RUN pip install -r /tmp/requirements.txt
+RUN apt-get update && \
```
Comment (jpuerto-psc) [inline docker/Dockerfile:10]: You can remove the sssd install here, it's unique to user-workspaces

File: docker/user_templates_api/nginx/conf.d/passthrough.conf [hunk 2]
```diff
@@ -0,0 +1,14 @@
+server {
```
Comment (jpuerto-psc) [inline docker/user_templates_api/nginx/conf.d/passthrough.conf:1]: Can remove this file

File: requirements.txt [hunk 3]
```diff
@@ -1,23 +1,28 @@
 aiohttp==3.8.3
```
Comment (jpuerto-psc) [inline requirements.txt:1]: Go through and remove django-q references

File: src/user_templates_api/settings.py [hunk 4]
```diff
@@ -126,3 +128,9 @@
 # https://docs.djangoproject.com/en/3.1/howto/static-files/
 
 STATIC_URL = '/static/'
+
```
Comment (jpuerto-psc) [inline src/user_templates_api/settings.py:131]: Can remove

File: src/user_templates_api/settings.py [hunk 5]
```diff
@@ -46,6 +47,7 @@
     'django.contrib.sessions',
     'django.contrib.messages',
     'django.contrib.staticfiles',
+    'django_q',
```
Comment (jpuerto-psc) [inline src/user_templates_api/settings.py:50]: Can remove

**Comment by gesinaphillips** [comment]
@jpuerto-psc when I start the docker web container it logs `INFO: ASGI 'lifespan' protocol appears unsupported.` Debugging with `--lifespan on` in start.sh reveals the following:

```
ERROR:    Exception in 'lifespan' protocol
uta-web-1    | Traceback (most recent call last):
uta-web-1    |   File "/usr/local/lib/python3.8/site-packages/uvicorn/lifespan/on.py", line 86, in main
uta-web-1    |     await app(scope, self.receive, self.send)
uta-web-1    |   File "/usr/local/lib/python3.8/site-packages/uvicorn/middleware/proxy_headers.py", line 78, in __call__
uta-web-1    |     return await self.app(scope, receive, send)
uta-web-1    |   File "/usr/local/lib/python3.8/site-packages/django/core/handlers/asgi.py", line 150, in __call__
uta-web-1    |     raise ValueError(
uta-web-1    | ValueError: Django can only handle ASGI/HTTP connections, not lifespan.
uta-web-1    | ERROR:    Application startup failed. Exiting.
```

**Review by jpuerto-psc — APPROVED** [review]

**Pull request merged**